File: src/main/java/fr/aiko/Karl/parser/Token.java
package fr.aiko.Karl.parser;

public class Token {
    private final int position;
    private final int line;
    private final TokenType type;
    private String value;

    public Token(TokenType type, String value, int position, int line) {
        this.type = type;
        this.value = value;
        this.position = position;
        this.line = line;
    }

    public TokenType getType() {
        return type;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String s) {
        value = s;
    }

    public int getPosition() {
        return position;
    }

    public int getLine() {
        return line;
    }
}

File: src/main/kotlin/studio/karllang/karl/parser/Token.kt
package studio.karllang.karl.parser

class Token(private val type: TokenType, private var value: String, private val position: Int, private val line: Int) {
    fun getType(): TokenType {
        return type
    }

    fun getValue(): String {
        return value
    }

    fun setValue(newValue: String) {
        value = newValue
    }

    fun getPosition(): Int {
        return position
    }

    fun getLine(): Int {
        return line
    }
}